staging/src/k8s.io/apiserver/pkg/server/options/server_run_options.go

```diff
@@ -337,5 +345,15 @@ func (s *ServerRunOptions) AddUniversalFlags(fs *pflag.FlagSet) {
 		"This option, if set, represents the maximum amount of grace period the apiserver will wait "+
 		"for active watch request(s) to drain during the graceful server shutdown window.")
 
+	utilfeature.DefaultMutableVersionedFeatureGate.DeferErrorsToValidation(true)
 	utilfeature.DefaultMutableFeatureGate.AddFlag(fs)
+	serverversion.Effective.AddFlags(fs)
+}
+
+// Complete sets the final emulation version for the feature gate.
+func (s *ServerRunOptions) Complete() error {
+	if utilfeature.DefaultFeatureGate.Enabled(genericfeatures.EmulationVersion) {
+		return utilfeature.DefaultMutableVersionedFeatureGate.SetEmulationVersion(serverversion.Effective.EmulationVersion())
```

Comment:
How can I use a different featuregate instance for this?  Since this `.Complete` is an explicit call, it seems as though it would be practical to accept the instance of `VersionedFeatureGate` you want to force to emulate and probably use the specified instanced to wire to the `--feature-gates` flag.